Dockerfile

```diff
@@ -1,18 +1,22 @@
 FROM golang:1.11-stretch AS builder
-WORKDIR /go/src/github.com/pusher/oauth2_proxy
+
+# Download tools
+RUN wget -O $GOPATH/bin/dep https://github.com/golang/dep/releases/download/v0.5.0/dep-linux-amd64
+RUN chmod +x $GOPATH/bin/dep
+
+# Copy sources
+WORKDIR $GOPATH/src/github.com/pusher/oauth2_proxy
 COPY . .
 
 # Fetch dependencies
-RUN wget -O dep https://github.com/golang/dep/releases/download/v0.5.0/dep-linux-amd64
-RUN chmod +x dep
-RUN mv dep $GOPATH/bin/dep
 RUN dep ensure --vendor-only
 
-# Build image
-RUN ./configure && make clean oauth2_proxy
+# Build binary
+RUN ./configure && make build.static && strip oauth2_proxy.static
```

Comment:
I'd like to suggest that we don't strip here at all. I've had a look and it only saves about 3mb (of a 12mb binary) but could make debugging harder if stack traces from panics are missing information that has been stripped (function names for instance)

I appreciate trying to optimise the image size but the bigger win here will be switching from debian to alpine.

Reply:
Thanks for the suggestions.  I've wrapped them up in 2 commits.